src/js/Component/State.js

```diff
@@ -77,7 +77,7 @@ Wui.stateMachine.prototype = {
                             stateStr += ((i > 0) ? '/' : '') + stateArray[i].view;
                             
                             for(var j = 0; j < keys.length; j++)
-                                stateStr += ((j > 0) ? '&' : '?') + keys[j] + '=' + stateArray[i].params[keys[j]];
+                                stateStr += ((j > 0) ? '&' : '?') + keys[j] + '=' + encodeURI(stateArray[i].params[keys[j]]);
```

Comment:
This is probably a great change, but a comment here on WHY it was made would be useful! My understanding is that in a URL everything after the `#` can be rather free-form, and that's why something like this wasn't originally included, so I'm very interested in the 'why'!